app/src/main/java/org/mozilla/firefox/vpn/main/vpn/GuardianVpnService.kt

```diff
@@ -18,9 +24,30 @@ class GuardianVpnService : WireGuardVpnService() {
     }
 
     override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        val turnOn = {
+            startForeground(NotificationUtil.DEFAULT_NOTIFICATION_ID, NotificationUtil.createBaseBuilder(this).build())
+        }
         when (intent?.getStringExtra(EXTRA_COMMAND)) {
-            COMMAND_TURN_ON -> startForeground(NotificationUtil.DEFAULT_NOTIFICATION_ID, NotificationUtil.createBaseBuilder(this).build())
-            COMMAND_TURN_OFF -> stopForeground(false)
+            COMMAND_TURN_OFF -> {
+                Log.d(logTag, "Handling 'turn off' command")
+                // Service being turned off manually by the user.
+                stopForeground(false)
+            }
+            COMMAND_TURN_ON -> {
+                Log.d(logTag, "Handling 'turn on' command")
+                // Service being started manually by the user.
+                turnOn()
+            }
+            null -> {
+                Log.d(logTag, "Handling system startup")
+                // If Always-On mode is turned on, OS will start us without any extra flags.
+                // Below we'll "properly" start ourselves, setting up a tunnel with a selected server, etc.
+                // We'll end-up circling back to this method, since while it's turning on, wireguard's
+                // TunnelManager ends up starting this service with a COMMAND_TURN_ON extra.
+                CoroutineScope(Dispatchers.IO).launch {
+                    this@GuardianVpnService.connectVpnIfPossible(logTag)
+                }
```

Comment:
Again, missing context: Would it make sense to scope this to the service's lifetime (onCreate(). onDestroy()) instead of creating a scope that is not managed (if that needs to be the case then I guess we could make it explicit use `GlobalScope`?)